python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -928,3 +925,32 @@ def evaluate_for_asset(
                 context.asset_key, context.partitions_def, rate_limited_asset_partitions
             ),
         )
+
+
+@whitelist_for_serdes
+class BlockingAssetCheckRule(AutoMaterializeRule, NamedTuple("_BlockingAssetcheckRule", [])):
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    def evaluate_for_asset(
+        self, context: AssetConditionEvaluationContext
+    ) -> "AssetConditionResult":
+        from dagster._core.storage.asset_check_execution_record import (
+            AssetCheckExecutionRecordStatus,
+        )
+
+        parent_assets = context.instance_queryer.asset_graph.get_parents(context.asset_key)
+        checks_on_parent_assets = [
+            key
+            for key in context.instance_queryer.asset_graph.blocking_asset_check_keys
+            if key.asset_key in parent_assets
+        ]
+        check_executions = context.instance_queryer.instance.event_log_storage.get_latest_asset_check_execution_by_key(
```

Comment:
One annoyance here is that this requires us to hit the database once per asset on every single tick.

If this is part of the default AssetCondition, and there are thousands of assets, then this can add a non-negligible amount of computation time to every single tick, as you'd need 1000 queries every single tick.

The ideal way of handling this would be to have some bit of information on the AssetRecord which keeps track of the latest storage id across any checks for that asset. This would require an update to `store_asset_check_event`, which would toss the new event id for that asset check on the asset record during the write transaction.

We get the AssetRecord basically "for free", and so it'd be quite easy to look across all of your parent assets and see if they had any asset checks execute since the previous `max_storage_id`. If not, no need to query anything at all.

Reply:
Interesting.

This storage method would also let us batch across assets, but seems like that'd take some work with how conditions are structured